bin/create-template.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -e
+
+############################################################
+TEMPLATE_NAME=`cat ../config/config.json | awk '/template_name/{print $2}'`
+INSTANCE_TEMPLATE_NAME=${TEMPLATE_NAME}
+INSTANCE_NAME=${TEMPLATE_NAME}
+ZONE=europe-west1-b
```

Comment:
---- CREATING NODES ----
ERROR: (gcloud.compute.instance-groups.managed.create) Could not fetch resource:
 - Invalid value for field 'resource.instanceTemplate': 'https://www.googleapis.com/compute/v1/projects/totemic-carrier-259013/global/instanceTemplates/ethtemplate'. Unable to create an instance from instanceTemplate ethtemplate in zone europe-west1-b: 
        Invalid value for field 'instance.networkInterfaces[0].accessConfigs[0].natIP': The specified external IP address '34.77.34.133' was not found in region 'europe-west1'.

Reply:
Make sure you test this template with the main script

Reply:
I tested it again and it still works fine for me. Please have a look at the attached screenshots (ethtemplate).

![screenshot1](https://user-images.githubusercontent.com/37183172/72758271-98be4a00-3bd2-11ea-92d3-b725d71e4e86.png)
![Screenshot 2020-01-20 at 22 14 49](https://user-images.githubusercontent.com/37183172/72758276-9a880d80-3bd2-11ea-8fad-8121b1db55e7.png)
![Screenshot 2020-01-20 at 22 15 02](https://user-images.githubusercontent.com/37183172/72758278-9bb93a80-3bd2-11ea-9335-a7129bb4c440.png)

The error you are getting is because in your case the template is trying to claim an unreserved IP, I don't know why. It would be good if someone else checks this as well, so I can know better about the problem. If you have any suggestions/ideas about this error, feel free to shoot them out.

Reply:
Your script works but the template created has something wrong. After executing your script I tested the VM with the SUT script and I got the same error as Karim, so I guess he was testing your solution with the SUT script.
![image](https://user-images.githubusercontent.com/12150712/72761871-b513b400-3bdd-11ea-972f-3af234686570.png)

Also the machines of this template still do not have http enabled and we need to enable them to access the blockchain outside GCP.

Reply:
I think it because I am stopping the VM once it is created, as mentioned in the task description. Maybe we can put the integration as a new/follow-up task for the next sprint as this sprint is almost over. I will fix the http enabling and config.json parts to close this task

Reply:
Are you sure? I do not think so, the template needs a disk not a running VM. Actually the template we created weeks ago and the one we are using with the SUT works even if every machine is stopped.

Reply:
I don't see any other problem. What do you think is the issue here?
For now, try running only the create-template.sh script. It works fine and satisfies the requirements of the Issue created. The integration can be handled in a follow up task for the prerequisites script, as mentioned by Karim.

Reply:
Did you actually tested and look what your script is creating? 
It is still not opened for HTTP.
![image](https://user-images.githubusercontent.com/12150712/72811859-c7830180-3c60-11ea-8a17-a0d0f1ea8d4a.png)

Solution for the problem:
https://stackoverflow.com/questions/35590641/what-is-the-gcloud-command-to-allow-http-traffic-on-a-vm-instance-its-not-cre

I copy pasted the error in Google and found out that the error was because the VM you are copying has an IP that later on you are trying to use again, which happens because the VM is still running when creating the template, this is why we should stop the VM before creating a template from it, to avoid running conflicts.

Cause of the SUT failure:
https://stackoverflow.com/questions/52541786/invalid-value-for-field-instance-networkinterfaces0-accessconfigs0-natip
Solution: change the order of the commands.

The fixes of these errors are quite simple as you can see in my commit. Although I still get another silly error when executing the SUT script which I do not have time to fix.

I know I did not require to test the solution with every script but as a group we need to work towards the same goal and test correctly what may be affected even though is not specified in the task. We can not create an overhead of tasks to fix tiny things just because the PRs were not tested in depth.

Reply:
True and thank you for the quick fixes. Let me have a look at the final silly error and then you can approve this PR.